### ISSUE TITLE
Create radio UVW frame

### ISSUE THREAD

**Issue by mhvk** [astropy/astropy#7766]
An attempt was made in #6882, but that may have been incomplete - see https://github.com/astropy/astropy/pull/6882#issuecomment-411385141 in particular

**Comment by adrn** [comment]
Just to re-raise my comment from #6882 with a suggestions: perhaps we could name the frame something like `RadioUVW` to distinguish it from the Galactic velocity components (also UVW)?

**Comment by mhvk** [comment]
@adrn - OK, I see, it is the cartesian representation of `Galactic` you are worried about. So, let's aim for `InterferometricUVW` (unless a better name appears); I've changed the title accordingly.

**Comment by eteq** [comment]
What about just `Interferometric`?  That actually seems cleaner to me - after all we don't call it `GalacticUVW` or even `GalacticLatLon`... Really `AltAz` is the only one with the names (and that's already gotten us into trouble... see #7708)

**Comment by cmurray-astro** [comment]
As the UVW coordinate system for interferometers is used to describe complex visibilities, how about `InterferometricVisibility`?

**Comment by eteq** [comment]
I like `InterferometricVisibility` - agree, @adrn and @mhvk?

**Comment by Joshuaalbert** [comment]
@mhvk If you want to still implement this, I have a reference that will speed things up. The Event Horizon Telescope (VLBI) calculates their UV coordinates [here](https://github.com/achael/eht-imaging/blob/master/ehtim/observing/obs_helpers.py#L55). They only define the U and V coordinates, and the W can be found easily using projU and projV in that source as ```projW = eye(3) - projU - projV```. However, I feel there is actually an easier and more robust way to do UVW with astropy's builtin coords.

Also, I'm not 100% certain the EHT's implementation is correct. I don't think it takes into account the motion of the observer which is different for antennas in different places on Earth.

**Comment by Joshuaalbert** [comment]
@caseyjlaw I found a good person to bring in, who will help get this implemented properly. Frits (@tikk3r) is a PhD student working with LOFAR including the international baselines. We don't call it VLBI because the baselines are only a few 1000km but it's still in the realm of VLBI. Anyways, Frits offers his attention to this. I think we can get it going in a few weeks. @tikk3r [this](https://github.com/astropy/astropy/pull/9869) is the PR related to it and we'll do our conversation there.

**Comment by Joshuaalbert** [comment]
@mhvk a few questions are needed to assert the accuracy of this implementation, based on astropy coordinates.
1. to what accuracy is the transform to/from GCRS to Barycentric coordinates performed in astropy? 1mm would be using:
```
r_bcrs = r_gcrs (1 - U_earth/c^2) - 1/2 (V_earth_bcrs . r_gcrs) V_earth_bcrs
``` 
Can you point me the right documentation, or code? Is there a way to control this accuracy?

2. Are the GCRS coordinates Terrestrial Time (TT) compatible i.e. linearly scaled so that time matches the rate of proper time on the geoid (the rate that atomic clocks would tick at stations on the Earth's surface)? Or are the geocentric coordinate time (TCG)?

3. Are the BCRS coordinates (i.e. ICRS) barycentric dynamical time (TDB) compatible? or barycentric coordinate time (TCB) compatible?

2 and 3 are important because depending on the convention used in astropy the coordinates may require a scaling to be correct at the VLBI level.

**Comment by mhvk** [comment]
@Joshuaalbert - I'm not 100% sure I understand what your `r` are for; the observer or the source?

Anyway, probably good to lay out the basics regardless: astropy internally uses SOFA routines (via ERFA). Note that we still do not have a proper BCRS, but the GCRS<->ICRS transformations are at https://github.com/astropy/astropy/blob/384e90b5132e386faaf168fdb1bda100d3eb26ef/astropy/coordinates/builtin_frames/icrs_cirs_transforms.py#L115

The basic idea is that we use https://pyerfa.readthedocs.io/en/latest/api/erfa.atciq.html and https://pyerfa.readthedocs.io/en/latest/api/erfa.atciqz.html with the astrometric parameters prefilled by https://pyerfa.readthedocs.io/en/latest/api/erfa.apcs.html (and Earth position/velocity either from built-in `epv00` or JPL ephemerides). The parallax shift is done "by hand". 

However, of course there are details. In reality we use slightly modified versions (for light bending near Sun); starting at https://github.com/astropy/astropy/blob/384e90b5132e386faaf168fdb1bda100d3eb26ef/astropy/coordinates/builtin_frames/utils.py#L169

And `apcs` is wrapped at https://github.com/astropy/astropy/blob/384e90b5132e386faaf168fdb1bda100d3eb26ef/astropy/coordinates/erfa_astrom.py#L93-L112

It seems that we pass in "TT" instead of the nominally required "TBD", though the apcs documentation states "TT can be used instead of TDB without any significant impact on accuracy." This may need to (and can) be revisted...

**Comment by Joshuaalbert** [comment]
Thank you for the in-depth reply. So I found "ICRS to GCRS covers space motion,  parallax, light deflection, and aberration. " Which is probably good to s few ps accuracy in delay modelling but I'll work on verifying this. What would be kind of cool is to have an option to control how precise to make the delay calculation.

**Comment by mhvk** [comment]
Agreed - I'm actually not quite sure sure still where that is done! But absolutely fine with changing things around (and try to get it upstream to SOFA if relevant).
